### ISSUE TITLE
`assert_type(A, Type[A])`: Expression is of type "Type[A]", not "Type[A]"

### ISSUE THREAD

**Issue by rsokl** [python/mypy#13623]
**Bug Report**

`assert_type(A, Type[A])` raises the error: Expression is of type "Type[A]", not "Type[A]"

**To Reproduce**

```python
# example.py
from typing import Type
from typing_extensions import assert_type

class A: ...

assert_type(A, Type[A])
```
```console 
$ mypy example.py 
example.py:7: error: Expression is of type "Type[A]", not "Type[A]"
Found 1 error in 1 file (checked 1 source file)
```
**Expected Behavior**

I don't think mypy should raise here

**Your Environment**

- Mypy version used: mypy 0.971 (compiled: yes)
- Mypy command-line flags: None
- Mypy configuration options from `mypy.ini` (and other config files): None
- Python version used: 3.9.13
- Operating system and version: Windows 11

**Comment by AlexWaygood** [comment]
Ew.

**Comment by sobolevn** [comment]
I can reproduce this on 0.971
But, looks like it is fixed on `master`, try this on: https://mypy-play.net/?mypy=master&python=3.10

I am going to close this issue, please reopen if this problem remains.
Thanks for the report!

**Comment by hauntsaninja** [comment]
I think this was fixed by #13311
